### PR TITLE
fix(cli codegen): fixed missing imports 

### LINKED ISSUE #2969: generating a command line : undefined: cli
## Problem statement

```bash
matteo@debian:~/go/src$ go build cmd/talendcloud/main.go 
# example/mymodule/cli
cli/engine_model.go:118:6: registerEngineClusterID redeclared in this block
	cli/engine_cluster_model.go:148:6: other declaration of registerEngineClusterID
cli/engine_model.go:529:6: retrieveEngineClusterIDFlags redeclared in this block
	cli/engine_cluster_model.go:439:6: other declaration of retrieveEngineClusterIDFlags
cli/runtime_run_profile_model.go:101:6: registerRuntimeRunProfileID redeclared in this block
	cli/runtime_model.go:107:6: other declaration of registerRuntimeRunProfileID
cli/runtime_run_profile_model.go:367:6: retrieveRuntimeRunProfileIDFlags redeclared in this block
	cli/runtime_model.go:281:6: other declaration of retrieveRuntimeRunProfileIDFlags
cli/artifact_model.go:149:47: undefined: models
cli/artifact_model.go:185:44: undefined: models
cli/artifact_model.go:213:46: undefined: models
cli/artifact_model.go:241:46: undefined: models
cli/artifact_model.go:269:50: undefined: models
cli/artifact_model.go:283:51: undefined: models
cli/artifact_model.go:283:51: too many errors
```

## Environment
swagger: https://api.us.cloud.talend.com/tmc/swagger/api-v2.6.json
go version: 21
OS:   Debian testing

### PR DESCRIPTION
**Pull request by fredbi**
* fixes #2969

This PR addresses an issue with goimports not being able
to find another generated package.

When generating files in 2 packages simultaneously (i.e. cli imports models
and files are being generated model per model in both packages),
there are situations when goimports fails to properly resolve the imported
package (i.e. the imported package is not completed and may have errors
while generating).

This fix proceeds in 2 steps: first generate all models, then proceed with
a final generation step ("PostModels") that iterates again over all
models. It is a bit slower, but imports are now safe.

Other additions with this PR:
* updated dependency to golang.org/x/tools
* added go-openapi/strfmt to default imports (was missing and added by
  goimports, but it is best to make it explicit and avoid unnecessary
  import fixes)
* in templates, added standard library imports
* relinted code generated by the CLI templates (check for errors, unused
  args, shadowed variables, prefer // over block comments, blank lines...)
* updated cli.gotmpl to use os.UserConfigDir() and os.HomeDir() rather
  than importing a third party package
* added guard for multiline .Description in calls to PersistentFlags()

Tests:
* added CLI codegen tests for https://github.com/go-swagger/go-swagger/issues/2969, https://github.com/go-swagger/go-swagger/issues/2650 and the latest docker spec
  (used to generate go-swagger/dockerctl)

Signed-off-by: Frederic BIDON <fredbi@yahoo.com>

### CODE REVIEW

**Comment by fredbi** [comment]
Built on top of #3045

**Review by casualjim — APPROVED** [review]

File: generator/templates/cli/cli.gotmpl [hunk 1]
```diff
@@ -138,14 +146,24 @@ func initViperConfigs() {
...
-	}else{
-		// look for default config
-		// Find home directory.
-		home, err := homedir.Dir()
-		cobra.CheckErr(err)
-
-		// Search config in home directory with name ".cobra" (without extension).
-		viper.AddConfigPath(path.Join(home, ".config", exeName))
+	} else{
+    var (
```
Comment (youyuanwu) [inline generator/templates/cli/cli.gotmpl:150]: nit: indent

File: generator/templates/cli/cli.gotmpl [hunk 2]
```diff
@@ -138,14 +146,24 @@ func initViperConfigs() {
...
+      configDir string
+      err error
+    )
+
+		// look for default config (OS-specific, e.g. ".config" on linux)
+		configDir, err = os.UserConfigDir()
+    if err != nil {
```
Comment (youyuanwu) [inline generator/templates/cli/cli.gotmpl:157]: nit: indent

**Review by youyuanwu — APPROVED** [review]

File: generator/templates/cli/cli.gotmpl (hunk 1 above)
Comment (fredbi) [reply generator/templates/cli/cli.gotmpl:154]: @youyuanwu ok done. Thxs

**Pull request merged**